### PR TITLE
Writer changes for Variant Stats

### PR DESCRIPTION
**Pull request by awenocur**
[sc-44861]

### CODE REVIEW

**Comment by awenocur** [comment]
Check that the variant stats array version is bumped.

**Comment by awenocur** [comment]
[sc-44861]

File: libtiledbvcf/src/stats/variant_stats.cc [hunk 1]
```diff
@@ -435,7 +456,7 @@ void VariantStats::process(
       return;
     }
   } else {
-    return;  // ngt <= 0
+    return;  // ngt >= 0
```
Comment (gspowley) [inline libtiledbvcf/src/stats/variant_stats.cc:459]: Let's remove the comment. The return actually happens when `ngt < 0`, which is easy enough to see from the code.

**Review by gspowley — COMMENTED** [review]
Looks good so far! A couple of more items to complete are captured in shortcut.

File: libtiledbvcf/src/stats/variant_stats.cc (hunk 1 above)
Comment (awenocur) [reply libtiledbvcf/src/stats/variant_stats.cc:467]: whoops; I misinterpreted that as one of the clang-format generated end bracket comments

File: libtiledbvcf/src/stats/variant_stats.cc [hunk 2]
```diff
@@ -463,16 +492,39 @@ void VariantStats::process(
...
     if (!gt_missing[i]) {
       auto alt = alt_string(ref, rec->d.allele[gt[i]]);
 
       if (gt[i] == 0) {
-        values_["ref"][AC] += count_delta_;
+        std::string ref_key = "ref";
+        if (is_nr_block) {
+          // ref block
+          ref_key = "nr";
+          values_["nr"][CUM_MAX] = max_length_;
```
Comment (gspowley) [inline libtiledbvcf/src/stats/variant_stats.cc:519]: Is this case, max_length will be the max length of ref blocks. Should we keep track of the longest record, including variants and ref blocks?

File: libtiledbvcf/src/stats/variant_stats.h [hunk 3]
```diff
@@ -206,16 +206,17 @@ class VariantStats {
...
   inline static const std::vector<std::string> COLUMN_STR = {
-      "contig", "pos", "allele"};
+      "contig", "pos", "sample", "allele"};
 
   // Array attributes
-  enum Attr { AC = 0, N_HOM, LAST_ };
-  inline static const std::vector<std::string> ATTR_STR = {"ac", "n_hom"};
+  enum Attr { AC = 0, AN, N_HOM, CUM_MAX, END_POS, LAST_ };
+  inline static const std::vector<std::string> ATTR_STR = {
+      "ac", "an", "n_hom", "cum_max", "end_pos"};
```
Comment (gspowley) [inline libtiledbvcf/src/stats/variant_stats.h:219]: Let's rename `cum_max` to `max_length` and `end_pos` to `end`.

**Review by gspowley — APPROVED** [review]
Looks good!

**Pull request merged**